src/estimation.jl

```diff
@@ -38,4 +40,166 @@ function esprit(x::AbstractArray, M::Integer, p::Integer, Fs::Real=1.0)
+
+Estimate the largest frequency (in Hz) in the signal `x` using Jacobsen's
+algorithm [^Jacobsen2007].
+
+[^Jacobsen2007]: E Jacobsen and P Kootsookos, "Fast, Accurate Frequency Estimators", Chapter
+10 in "Streamlining Digital Signal Processing", edited by R. Lyons, 2007, IEEE Press.
+"""
+function jacobsen(x::Vector{<:Real}, Fs::Real = 1.0)
+    N = length(x)
+    X = rfft(x)
+    k = argmax(abs.(X))  # index of DFT peak
+    if (k+1 <= N) && (k-1 >= 1)
+        δ = -real((X[k+1] - X[k-1]) / (2X[k] - X[k-1] - X[k+1]) )
+    else
+        δ = 0.0
+    end
+
+    (k + δ)*Fs/N
+end
+
+function jacobsen(x::Vector{<:Complex}, Fs::Real = 1.0)
+    N = length(x)
+    X = fftshift(fft(x))
+    k = argmax(abs.(X))  # index of DFT peak
+    f = fftshift(fftfreq(N, Fs))[k]  # frequency at index k
+    if (k+1 <= N) && (k-1 >= 1)
+        δ = -real((X[k+1] - X[k-1]) / (2X[k] - X[k-1] - X[k+1]) )
+    else
+        δ = 0.0
+    end
+
+    f + δ*Fs/N
+end
+
+"""
+    quinn(x::Vector, f0::Real = 0.0, Fs::Real = 1.0 ; tol = 1e-6, maxiters = 20)
+
+    quinn(x::Vector, Fs::Real = 1.0 ; kwargs...)
+
+    quinn(x::Vector ; kwargs...)
+
+Algorithms by Quinn and Quinn & Fernandes for frequency estimation. Given a
+signal `x` and an initial guess `f0`, estimate and return the frequency of the
+largest sinusoid in `x`. `Fs` is the sampling frequency. All frequencies are
+expressed in Hz.
+
+If the initial guess `f0` is not provided or if it is equal to zero, then a guess
+is calculated by `quinn` using Jacobsen's estimator. The sampling frequency `Fs`
+defaults to `1.0`.
+
+The following keyword arguments control the algorithm's behavior:
+
+- `tol`: () the algorithm stops when the absolut value of the
+  difference between two consecutive estimates is less than `tol`. Defaults to
+  `1e-6`.
+- `maxiters`: the maximum number of iterations to run. Defaults to `20`.
+
+Returns a tuple `(estimate, reachedmaxiters)`, where `estimate` is the
+estimated frequency, and `reachedmaxiters` is `true` if the algorithm finished
+after running for `maxiters` iterations (this may indicate that the algorithm
+did not converge).
+
+If the signal `x` is real, then the algorithm used is [^Quinn]. If the signal is
+complex, the algorithm is [^Quinn2009].
+
+[^Quinn1991]: B Quinn and J Fernandes, "A fast efficient technique for the
+estimation of frequency", Biometrika, Vol. 78 (1991).
+
+[^Quinn2009]: B Quinn, "Recent advances in rapid frequency estimation", Digital
+Signal Processing, Vol. 19 (2009), Elsevier.
+
+"""
+quinn(x ; kwargs...) = quinn(x, 0.0, 1.0 ; kwargs...)
+
+quinn(x, Fs ; kwargs...) = quinn(x, 0.0, Fs ; kwargs...)
+
+function quinn(x::Vector{<:Real}, f0::Real, Fs::Real ; tol = 1e-6, maxiters = 20)
+    fₙ = Fs/2
+    T = length(x)
+
+    # Run a quick estimate of largest sinusoid in x
+    if f0 == 0.0
+        f_est = jacobsen(x, Fs)
+        ω̂ = π*f_est/fₙ  # estimate ranges from 0 to π
+    else
+        ω̂ = π*f0/fₙ
+    end
```

Comment:
```suggestion
    ω̂ = π*f0/fₙ
```
(with above changes to the other methods)